.github/workflows/test.yml

```diff
@@ -0,0 +1,69 @@
+name: Test
+
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches-ignore:
+      - master
+    tags-ignore:
+      - v*
+
+jobs:
+  test:
+    name: Test ${{ matrix.os }} ${{ matrix.lib }} ${{ matrix.release_type }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      matrix:
+        # We only run on a single os version to not run into CI limits.
+        os: [
+          # ubuntu-16.04, # builds on this version are not picked up quickly on the free plan
+          # ubuntu-18.04,
+          ubuntu-20.04,
+          # macos-10.5, # builds on this version are not picked up quickly on the free plan
+          macos-11.0,
+          # macos-12.00, # builds on this version are not picked up quickly on the free plan
+          # windows-2019,
+          windows-2022
+        ]
+        lib: [
+          cxxlib,
+          python3lib,
+          nodejslib,
+          emscriptenlib
+        ]
+        release_type: [debug, release]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Install & Build
+        shell: bash
+        run: |
+          if [[ "${{ matrix.os }}" == "windows"* ]]; then
+            export PYTHON_EXECUTABLE="C:/hostedtoolcache/windows/Python/3.11.0/x64/python3.exe"
```

Comment:
I also added a PYTHON_EXECUTABLE option to the install-{platform}.sh script, so it will compile Boost.Python for the given python executable version / include / lib paths.